gateway-ha/src/main/java/io/trino/gateway/ha/persistence/dao/SelectorsDao.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.persistence.dao;
+
+import io.trino.gateway.ha.router.ResourceGroupsManager;
+import org.jdbi.v3.sqlobject.customizer.BindBean;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.util.List;
+
+public interface SelectorsDao
+{
+    @SqlQuery("""
+            SELECT * FROM selectors
+            """)
+    List<Selectors> findAll();
+
+    @SqlQuery("""
+            SELECT * FROM selectors
+            WHERE resource_group_id = :resourceGroupId
+            """)
+    List<Selectors> findByResourceGroupId(long resourceGroupId);
+
+    @SqlQuery("""
+            SELECT * FROM selectors
+            WHERE
+                resource_group_id = :resourceGroupId
+                AND priority      = :priority
+                AND user_regex    IS NOT DISTINCT FROM :userRegex
+                AND source_regex  IS NOT DISTINCT FROM :sourceRegex
+                AND query_type    IS NOT DISTINCT FROM :queryType
+                AND client_tags   IS NOT DISTINCT FROM :clientTags
+                AND selector_resource_estimate IS NOT DISTINCT FROM :selectorResourceEstimate
+            LIMIT 1
+            """)
+    Selectors findFirst(@BindBean ResourceGroupsManager.SelectorsDetail selector);
```

Comment:
To be honest I dont like this pattern .. wouldnt it be better to have a unique id for each . and find by Id .. 

And yes .. I realize that would be a change of the data model and should probably be a separate PR..  just wanted to bring it up since this really should only return one record .. but nothing seems to really enforce that on the db level

Reply:
The `selectors` table is defined in Trino.
We have to update the DDL in Trino and carefully avoid breaking backward compatibility.
https://github.com/trinodb/trino/blob/53ad89113b7d1fdbafc6f89a0a75452c14300e1f/plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/db/ResourceGroupsDao.java#L71-L81

Reply:
Ok.. thats a separate large step .. definitely dont want to worry about that now.